pkg/pool-weighted/contracts/lbp/LiquidityBootstrappingPool.sol

```diff
@@ -387,4 +733,43 @@ contract LiquidityBootstrappingPool is BaseWeightedPool, ReentrancyGuard {
 
         return scalingFactors;
     }
+
+    // Recovery Mode
+
+    /**
+     * @notice Returns whether the pool is in Recovery Mode.
+     */
+    function inRecoveryMode() public view override returns (bool) {
+        return _poolState.decodeBool(_RECOVERY_MODE_BIT_OFFSET);
+    }
+
+    /**
+     * @dev Sets the recoveryMode state, and emits the corresponding event.
+     */
+    function _setRecoveryMode(bool enabled) internal virtual override {
+        _poolState = _poolState.insertBool(enabled, _RECOVERY_MODE_BIT_OFFSET);
+
+        emit RecoveryModeStateChanged(enabled);
+    }
```

Comment:
```suggestion
    /**
     * @dev Sets the recoveryMode state.
     */
    function _setRecoveryMode(bool enabled) internal virtual override {
        _poolState = _poolState.insertBool(enabled, _RECOVERY_MODE_BIT_OFFSET);
    }
```

`RecoveryModeStateChanged` is emitted by `RecoveryMode` now so we don't need to emit it here.